back/app/controllers/web_api/v1/permissions_custom_fields_controller.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+class WebApi::V1::PermissionsCustomFieldsController < ApplicationController
+  skip_after_action :verify_policy_scoped
+  before_action :set_permissions_custom_field, only: %i[show update destroy]
+
+  def index
+    authorize PermissionsCustomField.new(permission: permission)
+    permissions_custom_fields = permission.permissions_custom_fields.includes(:custom_field)
+    permissions_custom_fields = paginate permissions_custom_fields
+
+    render json: linked_json(
+      permissions_custom_fields,
+      WebApi::V1::PermissionsCustomFieldSerializer,
+      params: fastjson_params,
+      include: %i[custom_field]
+    )
+  end
+
+  def show
+    render json: WebApi::V1::PermissionsCustomFieldSerializer.new(
+      @permissions_custom_field,
+      params: fastjson_params,
+      include: %i[custom_field]
+    ).serialized_json
+  end
+
+  def create
+    permissions_custom_field = PermissionsCustomField.new({ permission: permission }.merge(permission_params_for_create))
+    authorize permissions_custom_field
+    SideFxPermissionsCustomFieldService.new.before_create permissions_custom_field, current_user
+    if permissions_custom_field.save
+      SideFxPermissionsCustomFieldService.new.after_create permissions_custom_field, current_user
+      render json: WebApi::V1::PermissionsCustomFieldSerializer.new(
+        permissions_custom_field,
+        params: fastjson_params
+      ).serialized_json, status: :created
+    else
+      render json: { errors: permissions_custom_field.errors.details }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    @permissions_custom_field.assign_attributes permission_params_for_update
+    authorize @permissions_custom_field
+    SideFxPermissionsCustomFieldService.new.before_update @permissions_custom_field, current_user
+    if @permissions_custom_field.save
+      SideFxPermissionsCustomFieldService.new.before_update @permissions_custom_field, current_user
+      render json: WebApi::V1::PermissionsCustomFieldSerializer.new(
+        @permissions_custom_field,
+        params: fastjson_params
+      ).serialized_json, status: :ok
+    else
+      render json: { errors: @permissions_custom_field.errors.details }, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    SideFxPermissionsCustomFieldService.new.before_destroy @permissions_custom_field, current_user
+    permissions_custom_field = @permissions_custom_field.destroy
+    if permissions_custom_field.destroyed?
+      SideFxPermissionsCustomFieldService.new.after_destroy permissions_custom_field, current_user
+      head :ok
+    else
+      head :internal_server_error
+    end
+  end
+
+  private
+
+  def set_permissions_custom_field
```

Comment:
@jamesspeake I changed it like this, as this is more in line with other controllers (e.g. for official feedback, comments...). There is an old tech debt ticket about revising common code patterns across all controllers.